stdlib/public/runtime/Demangle.cpp

```diff
@@ -359,6 +373,25 @@ static bool _buildDemanglingForGenericArgs(
       break;
     }
 
+    case GenericParamKind::Value: {
```

Comment:
How back-deployable do we expect integer generic parameters to be? Should there be a corresponding update to `getRuntimeVersionThatSupportsDemanglingType` so that we can instantiate metadata using integer parameters with open-coded accessors when targeting older runtimes that don't have demangler support?

Reply:
Oh that's a good point. I don't expect value generics to be back deployable at all for types which means functions generic over values can't reference them at all because they need to be bound to a generic type that has a value parameter. Effectively, value generics are runtime gated just like parameter packs for types were. I will add that compiler diagnostic which should hopefully resolve this particular comment because all types with value generics should have a runtime that supports them.